haystack/document_store/sql.py

```diff
@@ -16,42 +16,43 @@
 class ORMBase(Base):
     __abstract__ = True
 
-    id = Column(String, default=lambda: str(uuid4()), primary_key=True)
+    id = Column(String(100), default=lambda: str(uuid4()), primary_key=True)
     created = Column(DateTime, server_default=func.now())
     updated = Column(DateTime, server_default=func.now(), server_onupdate=func.now())
 
 
 class DocumentORM(ORMBase):
     __tablename__ = "document"
 
-    text = Column(String, nullable=False)
-    index = Column(String, nullable=False)
-    vector_id = Column(String, unique=True, nullable=True)
+    text = Column(Text, nullable=False)
+    index = Column(String(100), nullable=False)
+    vector_id = Column(String(100), unique=True, nullable=True)
 
     # speeds up queries for get_documents_by_vector_ids() by having a single query that returns joined metadata
     meta = relationship("MetaORM", backref="Document", lazy="joined")
 
+
 class MetaORM(ORMBase):
     __tablename__ = "meta"
 
-    name = Column(String, index=True)
-    value = Column(String, index=True)
-    document_id = Column(String, ForeignKey("document.id", ondelete="CASCADE"), nullable=False)
+    name = Column(String(100), index=True)
+    value = Column(Text, index=True)
```

Comment:
As per mysql [docs](https://dev.mysql.com/doc/refman/8.0/en/create-index.html) -
```
BLOB and TEXT columns also can be indexed, but a prefix length must be given.
```
I think one integration test with mysql will show these issue if any.

Reply:
I think we could revert to a larger string field than text as you suggested earlier. I tested that the tables get created without any errors on a MySQL instance.

Reply:
Yes I agree with you.